src/shared_utilities/drivers.jl

```diff
@@ -586,3 +587,70 @@ function initialize_drivers(
     radiation_drivers = initialize_drivers(r, coords)
     merge(atmos_drivers, radiation_drivers)
 end
+
+"""
+    make_update_drivers(a::Union{AbstractAtmosphericDrivers, Nothing},
+                          r::Union{AbstractRadiativeDrivers, Nothing},
+                         )
+
+Creates and returns a function which updates the atmospheric
+and radiative forcing variables ("drivers").
+"""
+function make_update_drivers(
+    a::Union{AbstractAtmosphericDrivers, Nothing},
+    r::Union{AbstractRadiativeDrivers, Nothing},
+)
+    update_atmos! = make_update_drivers(a)
+    update_radiation! = make_update_drivers(r)
+    function update_drivers!(drivers, t)
+        update_atmos!(drivers, t)
+        update_radiation!(drivers, t)
+    end
+    return update_drivers!
+end
+
+"""
+    make_update_drivers(d::Nothing)
+
+Creates and returns a function which updates the driver variables
+in the case of no driver variables.
+"""
+make_update_drivers(d::Nothing) = (p, t) -> nothing
+
+"""
+    make_update_drivers(a::PrescribedAtmosphere{FT}) where {FT}
+
+Creates and returns a function which updates the driver variables
+in the case of a PrescribedAtmosphere at a point.
+"""
+function make_update_drivers(a::PrescribedAtmosphere{FT}) where {FT}
+    function update_drivers!(drivers, t)
+        drivers.P_liq .= FT(a.liquid_precip(t))
```

Comment:
@Sbozzolo - what do you think about these update functions? drivers.field is a field, in the site level case it is a point field and being set to a single value. for global runs, we would have to modify this (maybe with another type than `PrescribedAtmosphere`?) to set the field in a different way. 

does this look flexible enough given how we will move away from calling a function at a point for computing the forcing at a particular time?

(these functions would be computed in a callback)

Reply:
I think that we would have something like `set_field!(drivers.T, a, t)` and internally do what needs to be done. 

We will probably need to move away from calling fields directly with `(t)`.